torchgeo/datasets/eurocrops.py

```diff
@@ -0,0 +1,285 @@
+        6, the prefix is the full code and the suffix is empty.
+    """
+    assert len(hcat_code) == 10
+    if hcat_level == 3:
+        return hcat_code[0:4], hcat_code[4:10]
+    elif hcat_level == 4:
+        return hcat_code[0:6], hcat_code[6:10]
+    elif hcat_level == 5:
+        return hcat_code[0:8], hcat_code[8:10]
+    elif hcat_level == 6:
+        return hcat_code, ""
+    raise ValueError(f"Invalid HCAT level {hcat_level}.")
+
+
+class EuroCrops(VectorDataset):
+    """EuroCrops Dataset (Version 9).
+
+    The `EuroCrops <https://www.eurocrops.tum.de/index.html>`__ dataset combines "all
+    publicly available self-declared crop reporting datasets from countries of the
+    European Union" into a unified format. The dataset is released under CC BY 4.0 Deed.
+
+    The dataset consists of shapefiles containing a total of X polygons. Each polygon
+    is tagged with a "EC_hcat_n" attribute indicating the harmonized crop name grown
+    within the polygon in the year associated with the shapefile.
+
+    If you use this dataset in your research, please follow the citation guidelines at
+    https://github.com/maja601/EuroCrops#reference.
+    """
+
+    base_url = "https://zenodo.org/records/8229128/files/"
+
+    hcat_fname = "HCAT2.csv"
+    hcat_md5 = "2443cb1ef05ad16840f0f6a4c685b72d"
+    # Valid HCAT levels.
+    hcat_levels = [3, 4, 5, 6]
+    # Name of the column containing HCAT code in CSV file.
+    hcat_code_column = "HCAT2_code"
+
+    label_name = "EC_hcat_c"
+
+    # Filename, year, and md5 of files in this dataset on zenodo.
+    zenodo_files = [
+        ("AT_2021.zip", 2021, "490241df2e3d62812e572049fc0c36c5"),
+        ("BE_VLG_2021.zip", 2021, "ac4b9e12ad39b1cba47fdff1a786c2d7"),
+        ("DE_LS_2021.zip", 2021, "6d94e663a3ff7988b32cb36ea24a724f"),
+        ("DE_NRW_2021.zip", 2021, "a5af4e520cc433b9014cf8389c8f4c1f"),
+        ("DK_2019.zip", 2019, "d296478680edc3173422b379ace323d8"),
+        ("EE_2021.zip", 2021, "a7596f6691ad778a912d5a07e7ca6e41"),
+        ("ES_NA_2020.zip", 2020, "023f3b397d0f6f7a020508ed8320d543"),
+        ("FR_2018.zip", 2018, "282304734f156fb4df93a60b30e54c29"),
+        ("HR_2020.zip", 2020, "8bfe2b0cbd580737adcf7335682a1ea5"),
+        ("LT_2021.zip", 2021, "c7597214b90505877ee0cfa1232ac45f"),
+        ("LV_2021.zip", 2021, "b7253f96c8699d98ca503787f577ce26"),
+        ("NL_2020.zip", 2020, "823da32d28695b8b016740449391c0db"),
+        ("PT.zip", 2021, "3dba9c89c559b34d57acd286505bcb66"),
+        ("SE_2021.zip", 2021, "cab164c1c400fce56f7f1873bc966858"),
+        ("SI_2021.zip", 2021, "6b2dde6ba9d09c3ef8145ea520576228"),
+        ("SK_2021.zip", 2021, "c7762b4073869673edc08502e7b22f01"),
+        # Year is unknown for Romania portion (ny = no year).
+        # ("RO_ny", ??),
+    ]
+
+    # Color palette to choose from.
+    # There are hundreds of classes so we pick color via modulo of class index.
```

Comment:
Does the dataset offer any recommended cmap? I know CDL does but don't know much about EuroCrops

Reply:
The closest I found is this but it just visualizes the classes on a color scale:
https://github.com/maja601/EuroCrops/blob/main/webinar_demo/visualization.ipynb

Reply:
Hmm, I don't love that either, but it's at least slightly more "official". Let's go with something similar to what they do. I'm afraid of cycling through a list because it means multiple classes will be mapped to the same color. The benefit of a floating point color scale is that it works even when the hcat level changes.

Reply:
OK I've switched to using continuous color map.